content/terraform/v1.12.x/docs/language/meta-arguments.mdx

```diff
@@ -0,0 +1,211 @@
+
+The provider developer determines resource-specific arguments, but all resources support meta-arguments that let you manage resources' infrastructure lifecycle, including destruction behavior, preventing destruction, and establishing dependencies between resources. Terraform provides the following meta-arguments.
+
+## `depends_on`
+
+The `depends_on` meta-argument establishes dependencies between resources that do not directly reference one another. Use the `depends_on` argument to explicitly set the order in which Terraform creates resources.
+
+### Tutorials
+
+- [Create resource dependencies](/terraform/tutorials/configuration-language/dependencies)
+
+### Examples
+
+- [Specify a dependency during validation](/terraform/language/block/check#resource-dependency) 
+- [Specify a dependency when querying a data source](/terraform/language/block/data#specify-a-dependency)
+- [Specify a dependency when calling a module](/terraform/language/block/module#specify-dependencies) 
+- [Specify a dependency when outputting values](/terraform/language/block/output#explicit-dependencies) 
+- [Specify a dependency during resource creation](/terraform/language/block/resource#specify-a-dependency) 
+
+### Guidance 
+
+ - [Add dependencies](/terraform/language/data-sources#dependencies) when querying a data source
+ - [Set dependencies](/terraform/language/modules/configuration) in your module configuration
+
+### References
+
+- [`depends_on` argument in `check` blocks](/terraform/language/block/check#depends_on) 
+- [`depends_on` argument in `data` blocks](/terraform/language/block/data#depends_on)
+- [`depends_on` argument in `ephemeral` blocks](/terraform/language/block/ephemeral#depends_on)
+- [`depends_on` argument in `module` blocks](/terraform/language/block/module#depends_on)
+- [`depends_on` argument in `output` blocks](/terraform/language/block/output#depends_on)
+- [`depends_on` argument in `resource` blocks](/terraform/language/block/resource#depends_on)
+
+## `count`
+
+By default, Terraform manages one infrastructure object for each `resource` block. Terraform also creates single instances of a module per `module` block. You can use the `count` argument in `resource` and `module` blocks to create and manage multiple instances of each without writing a separate block for each instance.
+
+### How to choose between `count` and `for_each`
+
+The [`for_each` meta-argument](#for_each) performs a similar function to `count`, but lets you reference different values for each instance you create. Use the `count` argument when you want to create nearly identical instances. You should use `for_each` when some instance arguments must have distinct values that can't be directly derived from an
+Integer index. You cannot use both a `count` and ` for_each` argument in the same `resource` or `module` block.
```

Comment:
(to avoid mentioning ephemeral and data too)
```suggestion
Integer index. You cannot use both a `count` and ` for_each` argument in the same block.
```